ast.ts

```diff
@@ -25,6 +25,9 @@ export type Stmt<A> =
   | {  a?: A, tag: "while", cond: Expr<A>, body: Array<Stmt<A>> }
   | {  a?: A, tag: "pass" }
   | {  a?: A, tag: "field-assign", obj: Expr<A>, field: string, value: Expr<A> }
+  | {  a?: A, tag: "for", name: Expr, index?: Expr, iterable: Expr, body: Array<Stmt<A>> }
```

Comment:
Should `name` be a string, not an `expr`? Eventually it might be something like an `Assignable` if we have destructuring in this position. But I don't think you can put, say, a field lookup in this position.

Reply:
I've changed the `name` field into a string.

Reply:
Oh, and these need to be `Expr<A>` (see failing build below). Sorry I missed that in the first review.

Reply:
Thanks for noticing! I've resolved the conflict with `main` branch and passed the checks.